crates/freighter-storage/src/s3_client.rs

```diff
@@ -178,6 +179,25 @@ impl S3StorageProvider {
         // transient and only happen briefly when initially standing up the service with EC stores
         bail!("successfully put object but saw NotFound on pull 3 times");
     }
+
+    async fn list_prefix(&self, index: &str) -> StorageResult<Vec<String>> {
+        let objects = self
+            .client
+            .list_objects()
+            .bucket(self.bucket_name.clone())
+            .prefix(index)
+            .send()
+            .await
+            .context("failed to list all files")?;
+
+        let keys = objects
+            .contents()
+            .iter()
+            .filter_map(|obj| obj.key().map(ToString::to_string))
```

Comment:
`ToString` is a weird internal trait that exists only because `core`/`std` split prevented it on `Display`.
 
`String::from`?